src/charm.py

```diff
@@ -300,6 +303,20 @@ def _start_opensearch(self) -> bool:
             self.unit.status = BlockedStatus(" - ".join(missing_sys_reqs))
             return False
 
+        # check if there are shards that are "busy" or "relocating"
+        # defer the start until all the shards are "started"
+        if self.app_peers_data.get("leader_ip"):
+            try:
+                busy_shards = ClusterState.busy_shards_by_unit(self.opensearch)
+                if busy_shards:
+                    message = WaitingForBusyShards.format(
+                        " - ".join([f"{key}/{','.join(val)}" for key, val in busy_shards.items()])
+                    )
+                    self.unit.status = BlockedStatus(message)
+                    return False
+            except OpenSearchHttpError:
+                pass
+
```

Comment:
hmm, so this indeed checks if other units are busy, to know whether it can be started (becoming "busy" on its own, as the shards syncing will only happen after the opensearch service is started) 
but since this specific unit is not started yet, it can only get this info from already started units, and the safest to query is the leader unit. 
I missed passing the leader unit host to the `busy_shards_by_unit` method.